android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SERPTelemetryJsonParser.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.toList
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import mozilla.components.support.remotesettings.readOrFetch
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+
+/**
+ * Parse SERP Telemetry json from remote config.
+ */
+
+class SERPTelemetryJsonParser(context: Context) {
+
+    private val rootStorageDirectory: File = context.filesDir
+    private val coroutineScope = CoroutineScope(Dispatchers.IO)
+    private val searchProviders: Deferred<List<SearchProviderModel>> =
+        coroutineScope.async {
+            remoteSettingsDownloader()
+        }
+
+    suspend fun getSearchProviders(): List<SearchProviderModel> {
+        return searchProviders.await()
+    }
+
+    /**
+     * Fetches remote data from storage. If unavailable, fetches from the remote server.
+     */
+    private suspend fun remoteSettingsDownloader(): List<SearchProviderModel> {
+        val jsonFields: MutableList<SearchProviderModel> = mutableListOf()
+
+        val remoteSettingsClient = RemoteSettingsClient(
+            serverUrl = "https://firefox.settings.services.allizom.org",
+            bucketName = "main",
+            collectionName = "search-telemetry-v2",
```

Comment:
We should consider turning these values into parameters that are passed into the class since these components are intended to be reusable. Here's an example: https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt#48

Reply:
We should also be able to use the production server now, so [looking at this](https://searchfox.org/mozilla-mobile/rev/549ba88df7d55199fa4dc31f133bbb5ebcbd40c0/firefox-android/android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt#35-37) I think that doesn't need to be defined here.